File: product/src/domain/models/product.py
from dataclasses import dataclass
from typing import Optional

from src.infrastructure.db.models.product import ProductORM


@dataclass
class Product:
    name: str
    short_description: str
    full_description: str
    composition: str
    weight: float
    price: float
    photo: str
    id: Optional[int] = None

    @classmethod
    def from_orm(cls, orm_prod: ProductORM) -> "Product":
        return cls(
            id = orm_prod.id,
            name = orm_prod.name,
            short_description = orm_prod.short_description,
            full_description = orm_prod.full_description,
            composition = orm_prod.composition,
            weight = orm_prod.weight,
            price = orm_prod.price,
            photo = orm_prod.photo,
        )

File: data/internal/domain/entity/product.go
package entity

import "time"

// Product представляет доменную модель продукта
type Product struct {
	ID               int     `json:"id"`
	Name             string  `json:"name"`
	ShortDescription string  `json:"short_description"`
	FullDescription  string  `json:"full_description"`
	Composition      string  `json:"composition"`
	Weight           float64 `json:"weight"`
	Price            float64 `json:"price"`
	Photo            string  `json:"photo"`
	CreatedAt        time.Time `json:"created_at"`
	UpdatedAt        time.Time `json:"updated_at"`
}

// ProductFilter представляет фильтры для поиска продуктов
type ProductFilter struct {
	ID   *int
	Name *string
}

